### PR TITLE
Added env for enabling the user to create public workflows.

### PR DESCRIPTION
**Pull request by Josephalexantony-aot**
# Issue Tracking

JIRA: 
Issue Type: BUG/ FEATURE
https://aottech.atlassian.net/browse/FWF-1747
# Changes

Added an env variable for creating public workflow. If the env variable is true, then the checkbox for creating the public workflow will be shown. in other case if the multitenancy is enabled, then user can create only private (tenant based) workflow 

# How has the change been tested?

# Screenshots (if applicable)

# Build Success screenshot (Till a CICD pipeline is set up)

# Notes

### CODE REVIEW

File: deployment/docker/README.md [hunk 1]
```diff
@@ -154,6 +154,7 @@ Variable name | Meaning | Possible values | Default value |
 ` DRAFT_ENABLED`|Enable draft feature|true/false
 `DRAFT_POLLING_RATE`|Control draft timing||15000
 `EXPORT_PDF_ENABLED`|Manage export to pdf feature|true/false
+`PUBLIC_WORKFLOW_ENABLED`|Create public or private workflow|true/false
```
Comment (sumesh-aot) [inline deployment/docker/README.md:157]: please change description to "Enable creating workflow for all tenants"

File: deployment/openshift/web_dc.yaml [hunk 2]
```diff
@@ -46,6 +46,7 @@ objects:
             "REACT_APP_DRAFT_ENABLED":"${REACT_APP_DRAFT_ENABLED}",
```
Comment (sumesh-aot) [inline deployment/openshift/web_dc.yaml:46]: I will create a ticket to remove these openshift templates

File: forms-flow-web/README.md [hunk 3]
```diff
@@ -66,6 +66,7 @@ is mentioned on the [link](../forms-flow-idm/keycloak/README.md#create-forms-flo
 ` DRAFT_ENABLED`|Enable draft feature|true/false
 `DRAFT_POLLING_RATE`|Control draft timing||15000
 `EXPORT_PDF_ENABLED`|Manage export to pdf feature|true/false
+`PUBLIC_WORKFLOW_ENABLED`|Create public or private workflow|true/false
```
Comment (sumesh-aot) [inline forms-flow-web/README.md:69]: Same description as above

File: forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js [hunk 4]
```diff
@@ -161,7 +164,11 @@ export default React.memo(
       // Deployment Source
       form.append("deployment-source", "Camunda Modeler");
       // Tenant ID
-      if (tenantKey && !applyAllTenants) {
+      if (tenantKey && !applyAllTenants && publicWorkflowEnabled === "true") {
```
Comment (sumesh-aot) [inline forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js:167]: Not an UI expert, but is it a good practice to use `===` with string value of boolean ?

**Review by sumesh-aot — APPROVED** [review]
Just some minor comments, looks good otherwise.

File: forms-flow-web/src/constants/constants.js [hunk 5]
```diff
@@ -54,7 +54,11 @@ export const KEYCLOAK_REALM =
   (window._env_ && window._env_.REACT_APP_KEYCLOAK_URL_REALM) ||
   process.env.REACT_APP_KEYCLOAK_URL_REALM ||
   "forms-flow-ai";
-export const KEYCLOAK_URL =
+  export const publicWorkflowEnabled =
```
Comment (sreehari-aot) [inline forms-flow-web/src/constants/constants.js:57]: make it all caps and refer previous boolean contstant declarations

File: CHANGELOG.md [hunk 6]
```diff
@@ -55,6 +55,7 @@ Mark  items as `Added`, `Changed`, `Fixed`, `Removed`, `Untested Features`, `Upc
 * Added environment variable `CUSTOM_SUBMISSION_URL`, `CUSTOM_SUBMISSION_ENABLED` for support form adapter.
 * Added environment variables `DRAFT_ENABLED`, `DRAFT_POLLING_RATE` to manage draft feature. 
 * Added environment variable `EXPORT_PDF_ENABLED`for pdf service.
+* Added environment variable `PUBLIC_WORKFLOW_ENABLED` for enabling public workflow creation
```
Comment (abhilash-aot) [inline CHANGELOG.md:58]: can mention this is for multitenant specific

File: forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js [hunk 7]
```diff
@@ -7,6 +7,7 @@ import { useTranslation } from "react-i18next";
 import { extractDataFromDiagram } from "../../helpers/helper";
 import { createXML } from "../../helpers/deploy";
 import { MULTITENANCY_ENABLED } from "../../../../constants/constants";
+import { publicWorkflowEnabled } from "../../../../constants/constants";
```
[the remark this answers is not in the document]
Comment (abhilash-aot) [reply forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js:10]: yes , please merge imports from the same path

File: forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js [hunk 8]
```diff
@@ -86,8 +87,11 @@ export default React.memo(
       );
     };
     useEffect(() => {
-      tenant === null ? setApplyAllTenants(true) : setApplyAllTenants(false);
-        if (diagramXML) {
+       if (publicWorkflowEnabled === "true") {
```
[the remark this answers is not in the document]
Comment (abhilash-aot) [reply forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js:90]: @Josephalexantony-aot Please check the draft/ pdf enabled feature addition. A constant to enable disable feature is added , add this to the same

File: deployment/openshift/web_dc.yaml (hunk 2 above)
Comment (Josephalexantony-aot) [reply deployment/openshift/web_dc.yaml:46]: okay

File: deployment/docker/README.md (hunk 1 above)
Comment (Josephalexantony-aot) [reply deployment/docker/README.md]: done

File: forms-flow-web/README.md (hunk 3 above)
Comment (Josephalexantony-aot) [reply forms-flow-web/README.md]: done

File: forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js (hunk 4 above)
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js:167]: done changes

File: forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js:10]: done

File: forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js:90]: done

File: forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js [hunk 9]
```diff
@@ -161,7 +164,11 @@ export default React.memo(
       // Deployment Source
       form.append("deployment-source", "Camunda Modeler");
       // Tenant ID
-      if (tenantKey && !applyAllTenants) {
+      if (tenantKey && !applyAllTenants && publicWorkflowEnabled === "true") {
+        form.append("tenant-id", tenantKey);
+      }
+      //If the env value is false,and Multitenancy is enabled, then by default it will create a tenant based workflow.
+      if (MULTITENANCY_ENABLED && publicWorkflowEnabled === "false") {
```
[the remark this answers is not in the document]
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js:171]: done

File: forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js [hunk 10]
```diff
@@ -341,9 +348,9 @@ export default React.memo(
         </div>
 
         <div>
-          {MULTITENANCY_ENABLED ? (
+          {MULTITENANCY_ENABLED && publicWorkflowEnabled === "true" ? (
```
[the remark this answers is not in the document]
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js:351]: done

File: forms-flow-web/src/components/Modeler/Editors/DmnEditor/index.js [hunk 11]
```diff
@@ -7,9 +7,9 @@ import { useTranslation } from "react-i18next";
 import { extractDataFromDiagram } from "../../helpers/helper";
 import { createXML } from "../../helpers/deploy";
 import { MULTITENANCY_ENABLED } from "../../../../constants/constants";
+import { publicWorkflowEnabled } from "../../../../constants/constants";
```
[the remark this answers is not in the document]
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Modeler/Editors/DmnEditor/index.js:10]: done

File: forms-flow-web/src/constants/constants.js (hunk 5 above)
Comment (Josephalexantony-aot) [reply forms-flow-web/src/constants/constants.js:57]: done

File: CHANGELOG.md (hunk 6 above)
Comment (Josephalexantony-aot) [reply CHANGELOG.md:58]: done

Comment (Josephalexantony-aot) [reply CHANGELOG.md]: done

File: forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js (hunk 4 above)
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js]: changed

File: forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (Josephalexantony-aot) [reply forms-flow-web/src/components/Modeler/Editors/BpmnEditor/index.js]: done

**Comment by Josephalexantony-aot** [comment]
Changes have been added

**Review by abhilash-aot — APPROVED** [review]

**Pull request merged**